build.gradle

```diff
@@ -150,6 +142,12 @@ project('common') {
         //Do NOT add any additional dependencies here.
     }
 
+    if (System.getProperty("buildJDK11Client") == null) {
```

Comment:
It's bit inconsistent to call this property this way as when default JDK is changed to the next version this property should be updated as well. Suggest to use a better name, something that will reflect the fact that Client will be build for the same release version as Server code.

Reply:
Also please don't forget to update repository README file to reflect the changes in default JDK

Reply:
Instead I added a user configurable property to set the client java version accordingly: `clientJavaVersion`. I opted for this instead of say, a `compileUsingSameJDK` flag because it essentially requires the same amount of user effort, but is more informative to what versions are being targeted.

Reply:
okay, makes sense.